server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -193,7 +197,7 @@ public static void encodeTsid(StreamOutput out, SortedMap<String, BytesReference
             Map<String, Object> result = new LinkedHashMap<String, Object>(size);
 
             for (int i = 0; i < size; i++) {
-                String name = in.readString();
+                String name = in.readBytesRef().utf8ToString();
```

Comment:
Here's the bug that made this not work - sometimes BytesRef and String don't write the same.

Reply:
I wonder if there is a value in hiding the fact that it's BytesRef all the way down and can this hurt us beyond serialization thing.

Reply:
Are you thinking this should be it's own class? Like one that just wraps `BytesRef` or something?

Reply:
Well, if I am completely honest, I was thinking why do we have 2 different maps with String in one case and BytesRef in another case that essentially represent the same data. But I think I like your interpretation much better. It has quite a bit of complexity that could probably benefit from some encapsulation.

Reply:
Just to be clear, maybe we need a TimeSeriesId class to represent this sort of data.

Reply:
Would you like me to do it in this or a followup?

Reply:
Yeah, let's not mix refactoring with the actual change.